docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,77 @@ UUID
    A specific UUID version is not required or guaranteed. This extension represents
    UUIDs as FixedSizeBinary(16) with big-endian notation and does not interpret the bytes in any way.
 
+Other
+=====
+
+Other represents a type or array that one Arrow-based system received from an
+external (likely non-Arrow) system, but cannot interpret itself.  In this
+case, the Other type explicitly communicates the name and presence of a field
+to downstream clients.
+
+For example:
+
+* A Flight SQL service may support connecting external databases.  In this
+  case, its catalog (``GetTables`` etc.) should reflect the names and types of
+  tables in external databases.  But those external systems may support types
+  it does not recognize.  Instead of erroring or silently dropping columns
+  from the catalog, it can use the Other[Null] type to report that a column
+  exists with a particular name and type name in the external database; this
+  lets clients know that a column exists, but is not supported.
+
+* The ADBC PostgreSQL driver, because of how the PostgreSQL wire protocol
+  works, may get bytes for a field whose type it does not recognize (say, a
+  geospatial type).  It can still return the bytes to the application which
+  may be able to parse the data itself.  In that case, it can use the
+  Other[binary] type to return the column data.  The Other type differentiates
+  the column from actual binary columns.
+
+Of course, the intermediate system *could* implement a custom extension type
+for these example types.  But there is no way in general that every type can
+be known in advance.  In such cases, the Other type allows the system to
+explicitly note that it does not support some type or field, without silently
+losing data or sending irrelevant errors.  It could also pretend to support
+the types by making up extension types on the fly.  But this misleads
+downstream systems who cannot tell if the type is supported or not.
+
+Extension parameters:
+
+* Extension name: ``arrow.other``.
+
+* The storage type of this extension is any type.  If there is no underlying
+  data, the storage type should be Null.  If there is data (because the system
+  got bytes or some other data it does not know how to interpret), the storage
+  type should preferably be binary or fixed-size binary, but may be any type.
```

Comment:
Is it useful to leave this open? Binary always would probably work in all situations.

Reply:
Though restricting to binary would probably not simplify things much...

Reply:
> I can think of at least one use case where binary may not be sufficient: DuckDB currently returns types defined in extensions as whatever the underlying DuckDB type is, which could be many arrow types

But why would we use Other in this case?

Reply:
I don't understand at all. Perhaps you can explain using a concrete example of information loss?

Reply:
Perhaps if DuckDB returns an extension type that is not registered at consumer side?

Reply:
But how would DuckDB know whether it is registered or not? And what would the "Other" type improve things exactly?

Reply:
Could consumer interpret received non-registered types as Other?

Reply:
Other is **not about the consumer**

Reply:
@paleolimbot your example of duckdb does not really fit under "Other represents a type or array that one Arrow-based system received from an external (likely non-Arrow) system, but cannot interpret itself."? 
In your example duckdb is the producer and is Arrow-aware, so shouldn't they just define an extension type themselves then, instead of using `arrow.other` ? (e.g. `duckdb.point2d`)

Reply:
I believe that's already addressed?

> I don't want to get hung up on DuckDB specifically (they might not be interested or might be able to implement the ability for runtime-loadable extensions to customize their Arrow export representation before they get a chance to implement this), I just wanted to demonstrate an example where the arbitrary payload that an ADBC driver (or general Arrow consumer) receives is not binary.

Reply:
Another example might be a PostgreSQL [user defined type](https://www.postgresql.org/docs/current/xtypes.html)? In principle we might know that it can be represented as a struct but may want to wrap in Other in case it's meant to have additional semantics (as in the example there - a Complex type which the driver might eventually want to properly map to an imaginary Arrow Complex extension type instead of leaving it as a bare struct)

Reply:
Complex is an example, as stated. In this case the converter would not know about it. Obviously, if it did know about it it would want to use the actual extension type. 

The point is that in the case that the converter does not know about the proper extension type, leaving it as the bare struct representation may be misleading and the converter at least knows that much from PostgreSQL - it would know there is a user defined type with a particular representation but not the intended Arrow representation. Other can communicate this explicitly, just leaving it as an Arrow struct would lose information.

Reply:
Ok, thanks for the explanation. Perhaps this example can be added to the spec as well :-)

Reply:
> it would have to do something like `f"duckdb.{type_name}"` to create the extension name, and I think we are trying to discourage that mechanism (i.e., only export an extension type if it has a type implementation somewhere)

Why would be discourage that? (I would assume that duckdb then also supports it upon import, so it would have an implementation somewhere)

Part of the rationale to explain the `arrow.other` canonical type in this PR now is that you should _not_ use it and still use a normal extension type if you know about the type. 

I used the `ogc.wkb` extension type before as example, asking whether GDAL should then also use this Other type instead. And it was said that that's an example where you should _not_ use it because GDAL perfectly knows that this is WKB data and can signal that to downstream consumers of its Arrow export APIs.

Reply:
Yeah, I appreciate the examples, I was trying to figure out a good example of a non-binary representation. I think I'll go with PostgreSQL composite types, though relying so heavily on Postgres for all examples also makes me easy